tf/mod_source_parse.go

```diff
@@ -115,7 +115,15 @@ func ParseSource(modsource string) (Source, error) {
 		rawURL := strings.TrimPrefix(modsource, "git::")
 		u, err := url.Parse(rawURL)
 		if err != nil {
-			return Source{}, errors.E(ErrInvalidModSrc, "invalid URL inside %s", modsource)
+			return Source{}, errors.E(ErrInvalidModSrc, modsource)
+		}
+
+		if u.Path == "" {
```

Comment:
Don't we have a test for this ? (usually it is quite easy to test the parsing scenarios here)